crates/fuel-core/src/graphql_api/storage/coins.rs

```diff
@@ -23,6 +41,180 @@ pub fn owner_coin_id_key(owner: &Address, coin_id: &UtxoId) -> OwnedCoinKey {
     default
 }
 
+/// The storage table for the index of coins to spend.
+
+// In the implementation of getters we use the explicit panic with the message (`expect`)
+// when the key is malformed (incorrect length). This is a bit of a code smell, but it's
+// consistent with how the `double_key!` macro works. We should consider refactoring this
+// in the future.
+pub struct CoinsToSpendIndex;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct CoinsToSpendIndexKey([u8; CoinsToSpendIndexKey::LEN]);
+
+impl Default for CoinsToSpendIndexKey {
+    fn default() -> Self {
+        Self([0u8; CoinsToSpendIndexKey::LEN])
+    }
+}
+
+impl core::fmt::Display for CoinsToSpendIndexKey {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "owner={}, asset_id={}, retryable_flag={}, amount={}",
+            self.owner(),
+            self.asset_id(),
+            self.retryable_flag(),
+            self.amount()
+        )
+    }
+}
+
+impl CoinsToSpendIndexKey {
+    const LEN: usize = Address::LEN
+        + AssetId::LEN
+        + u8::BITS as usize / 8
+        + u64::BITS as usize / 8
+        + TxId::LEN
+        + 2;
```

Comment:
I think it will be better if the type `u8::BITS as usize / 8` will be represented by `enum` with `repr(u8)`.

Instead of `u64::BITS as usize / 8` you can use `size_of::<Amount>`.

Instead of `TxId::LEN + 2` you could use `size_of::<UtxoId>`.

Reply:
Updated in d720d1db3569256c8b34e0fa1264701c8ac19290